integration_tests/version_generation_test.py

```diff
@@ -98,7 +98,29 @@ def extracted_sdist(
 
         with tempfile.TemporaryDirectory() as temp_dir:
             with tarfile.open(str(sdist_tarbal_path), 'r') as sdist_tarbal:
-                sdist_tarbal.extractall(temp_dir)
+                
+                import os
+                
+                def is_within_directory(directory, target):
```

Comment:
these helper function should go to some helper module so that these could be reused more easily